src/pybind/mgr/dashboard/frontend/src/app/shared/api/auth.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { fakeAsync, inject, TestBed, tick } from '@angular/core/testing';
+
+import 'rxjs/add/observable/of';
+import { Observable } from 'rxjs/Observable';
+
+import { AuthStorageService } from '../services/auth-storage.service';
+import { AuthService } from './auth.service';
+
+describe('AuthService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [AuthService, AuthStorageService],
+      imports: [HttpClientTestingModule]
+    });
+  });
+
+  it(
+    'should be created',
+    inject([AuthService], (service: AuthService) => {
+      expect(service).toBeTruthy();
+    })
+  );
+
+  it(
+    'should login and save the user',
+    inject(
+      [AuthService, HttpClient],
+      fakeAsync((service: AuthService, httpClient: HttpClient) => {
```

Comment:
Could you please inject the services inside a `beforeEach`? This way the tests are easier to read.

(Also in every other test file you have touched)

Reply:
Done